packages/postgres-database/src/simcore_postgres_database/utils_user_preferences.py

```diff
@@ -0,0 +1,65 @@
+import json
+from typing import Any
+
+import sqlalchemy as sa
+from aiopg.sa.connection import SAConnection
+from sqlalchemy.dialects.postgresql import insert as pg_insert
+
+from .models.user_preferences import user_preferences
+
+
+class CouldNotCreateOrUpdateUserPreferenceError(Exception):
+    ...
+
+
+class UserPreferenceNameHelper:
+    @classmethod
+    def get_preference_name(
+        cls, user_id: int, preference_name: str, product_name: str
+    ) -> str:
+        # NOTE: key is encoded with a minimal structure, allows for
+        # 100% certainty when decoding in the future if needed
+        return json.dumps([user_id, preference_name, product_name])
```

Comment:
why not simply add these as columns of user_preferences and use the triplet as primary key constrain? Similar to (service_key, version_key) unique constraint tuple. See https://github.com/ITISFoundation/osparc-simcore/blob/master/packages/postgres-database/src/simcore_postgres_database/models/services.py#L102

Reply:
I agree, also that means you can cascade deletion if the user/product disappears. also just plain searching in the db for debugging would be much simpler

Reply:
I've changed this, and we now have two separate tables for the two preference types that we currently support.